[Bug] cmake: imagery/i.zc/main.c:176: undefined reference to `del2g'

## Describe the bug

Building GRASS without FFTW support fails to build.

## To reproduce

1. Configure project

```
$ cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -DWITH_X11=no -DWITH_OPENGL=no -DWITH_CAIRO=no -DWITH_LIBPNG=yes -DWITH_SQLITE=no -DWITH_POSTGRES=no -DWITH_MYSQL=no -DWITH_ODBC=no -DWITH_ZSTD=yes -DWITH_BZLIB=yes -DWITH_READLINE=yes -DWITH_FREETYPE=yes -DWITH_NLS=yes -DWITH_FFTW=no -DWITH_CBLAS=no -DWITH_LAPACKE=no -DWITH_OPENMP=yes -DWITH_LIBSVM=no -DWITH_TIFF=yes -DWITH_NETCDF=no -DWITH_GEOS=no -DWITH_PDAL=no -DWITH_LIBLAS=OFF -DWITH_LARGEFILES=no -DWITH_DOCS=no -DWITH_GUI=no -DWITH_FHS=OFF -DCMAKE_BUILD_TYPE=RelWithDebInfo
```

2. Build

```
$ ninja
```

3. The build fails with

```
FAILED: [code=1] output/lib64/grass85/bin/i.zc
: && /usr/bin/x86_64-pc-linux-gnu-gcc -DGRASS_CMAKE_BUILD=1 -O2 -pipe -fno-strict-aliasing -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs -Wl,--dependency-file=imagery/CMakeFiles/i.zc.dir/link.d imagery/CMakeFiles/i.zc.dir/i.zc/main.c.o -o output/lib64/grass85/bin/i.zc  -Wl,-rpath,/var/tmp/portage/sci-geosciences/grass-8.5.0/work/grass-8.5.0_build/output/lib64/grass85/lib:  output/lib64/grass85/lib/libgrass_imagery.so.8.5.0  output/lib64/grass85/lib/libgrass_vector.so.8.5.0  output/lib64/grass85/lib/libgrass_raster.so.8.5.0  output/lib64/grass85/lib/libgrass_gmath.so.8.5.0  output/lib64/grass85/lib/libgrass_gis.so.8.5.0  -Wl,-rpath-link,/var/tmp/portage/sci-geosciences/grass-8.5.0/work/grass-8.5.0_build/output/lib64/grass85/lib && :
/usr/x86_64-pc-linux-gnu/binutils-bin/2.46.0/ld: imagery/CMakeFiles/i.zc.dir/i.zc/main.c.o: in function `main':
main.c:(.text.startup+0x4e4): undefined reference to `del2g'
collect2: error: ld returned 1 exit status
```

## Expected behavior

The software builds successfully.

## System description

- Operating System: Gentoo Linux
- GRASS version: `main` branch
- Build system: CMake
- Compiler: gcc (Gentoo 15.2.1_p20260214 p5) 15.2.1 20260214

## Additional context

Switching `-DWITH_FFTW=no` to `-DWITH_FFTW=yes` works.

Bug was reported through https://bugs.gentoo.org/974736.

--- a/imagery/CMakeLists.txt
+++ b/imagery/CMakeLists.txt
@@ -34,8 +34,11 @@ set(imagery_modules_list
     i.smap
     i.target
     i.topo.corr
-    i.vi
-    i.zc)
+    i.vi)
+
+if(WITH_FFTW)
+  list(APPEND imagery_modules_list i.zc)
+endif()
 
 if(WITH_LIBSVM)
   build_program_in_subdir(
@@ -220,14 +223,18 @@ build_program_in_subdir(
 build_program_in_subdir(i.vi DEPENDS grass_imagery grass_raster grass_vector
                         grass_gis ${LIBM})
 
-build_program_in_subdir(
-  i.zc
-  DEPENDS
-  grass_imagery
-  grass_raster
-  grass_vector
-  grass_gis
-  grass_gmath)
+if(WITH_FFTW)
+  build_program_in_subdir(
+    i.zc
+    DEPENDS
+    grass_imagery
+    grass_raster
+    grass_vector
+    grass_gis
+    grass_gmath)
+else()
+  message(STATUS "i.zc disabled because FFTW is not available")
+endif()
 
 build_program_in_subdir(
   i.fft

--- a/raster/CMakeLists.txt
+++ b/raster/CMakeLists.txt
@@ -105,7 +105,6 @@ set(raster_modules_list
     r.support.stats
     r.surf.area
     r.surf.contour
-    r.surf.fractal
     r.surf.gauss
     r.surf.idw
     r.surf.random
@@ -133,6 +132,10 @@ set(raster_modules_list
 
 if(WITH_LIBPNG)
   list(APPEND raster_modules_list r.in.png r.out.png)
+endif()
+
+if(WITH_FFTW)
+  list(APPEND raster_modules_list r.surf.fractal)
 endif()
 
 if(MSVC)
@@ -618,8 +621,12 @@ build_program_in_subdir(r.surf.area DEPENDS grass_gis grass_raster ${LIBM})
 
 build_program_in_subdir(r.surf.contour DEPENDS grass_gis grass_raster)
 
-build_program_in_subdir(r.surf.fractal DEPENDS grass_gis grass_raster
-                        grass_gmath ${LIBM})
+if(WITH_FFTW)
+  build_program_in_subdir(r.surf.fractal DEPENDS grass_gis grass_raster
+    grass_gmath ${LIBM})
+else()
+  message(STATUS "r.surf.fractal disabled because FFTW is not available")
+endif()
 
 build_program_in_subdir(r.surf.gauss DEPENDS grass_gis grass_raster grass_gmath)
 

